Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTraceViewModelTrace.swift

```diff
@@ -114,6 +67,63 @@ extension UtilityNetworkTraceViewModel {
     }
 }
 
+extension UtilityNetworkTraceViewModel.Trace {
+    /// - Parameter name: A name of a utility asset group.
+    /// - Returns: The set of utility elements returned by the trace that belong to the provided
+    /// asset group, grouped by type.
+    func elementsByTypeInGroup(named name: String) -> [String: [UtilityElement]] {
+        let assetsInGroup = elementsInAssetGroup(named: name)
+        var result = [String : [UtilityElement]]()
+        assetsInGroup.forEach { e in
+            var assetTypeGroup = result[e.assetType.name, default: []]
+            assetTypeGroup.append(e)
+            result.updateValue(assetTypeGroup, forKey: e.assetType.name)
+        }
+        return result
+    }
+    
+    /// - Parameter name: A name of a utility asset group.
+    /// - Returns: The set of utility elements returned by the trace that belong to the provided
+    /// asset group.
+    func elementsInAssetGroup(named name: String) -> [UtilityElement] {
+        return elementResults.filter({ $0.assetGroup.name == name })
+    }
+    
+    /// A set of the asset group names returned by the trace.
+    var assetGroupNames: Set<String> {
+        var assetGroupNames = Set<String>()
+        elementResults.forEach {
+            assetGroupNames.insert($0.assetGroup.name)
+        }
+        return assetGroupNames
+    }
+    
+    /// The extent of the trace's geometry result with a small added buffer.
+    var resultExtent: Envelope? {
+        guard let utilityGeometryTraceResult else { return nil }
+        
+        let geometries = [
+            utilityGeometryTraceResult.multipoint,
+            utilityGeometryTraceResult.polygon,
+            utilityGeometryTraceResult.polyline
+        ]
+            .compactMap { geometry in
+                if let geometry, !geometry.isEmpty {
+                    return geometry
+                } else {
+                    return nil
+                }
+            }
+        
+        guard let combinedExtents = GeometryEngine.combineExtents(of: geometries),
+              let expandedEnvelope = GeometryEngine.buffer(around: combinedExtents, distance: 200) else {
+            return nil
+        }
+        
+        return expandedEnvelope.extent
+    }
+}
+
 extension UtilityNetworkTraceViewModel.Trace: Equatable {
     static func == (
         lhs: UtilityNetworkTraceViewModel.Trace,
```

Comment:
FYI: Since `Trace` is a struct, `Self` can be used here, which makes the declaration fit on one line nicely:

```swift
static func == (lhs: Self, rhs: Self) -> Bool {
    return lhs.id == rhs.id
}
```